### DISCUSSION TITLE
Ability for Bito to understand your local codebase

### DISCUSSION THREAD

**Discussion by bitoandy** [gitbito/bitoai#86] [Announcements]
Hi Everyone, 

We are live on product hunt, where we are announcing the ability for Bito to understand your local codebase.  please check it out and vote: [https://www.producthunt.com/posts/bito-ai-chatgpt-for-your-codebase](https://www.producthunt.com/posts/bito-ai-chatgpt-for-your-codebase)

**Comment by MuhammadMahmoudH** [comment]
it doesn't work with a premium account Please, if you have a solution write it in steps

**Reply by bitoandy** [reply]
Hi @MuhammadMahmoudH ,

Checkout following links:
For VS Code: [https://docs.bito.ai/feature-guides/ai-that-understands-your-code/using-in-visual-studio-code](https://docs.bito.ai/feature-guides/ai-that-understands-your-code/using-in-visual-studio-code)
For JetBrains: [https://docs.bito.ai/feature-guides/ai-that-understands-your-code/using-in-jetbrains-ides](https://docs.bito.ai/feature-guides/ai-that-understands-your-code/using-in-jetbrains-ides)

**Reply by MuhammadMahmoudH** [reply]
This feature is no longer available for premium accounts on phpstorm, I will cancel the premium plan till the problem is fixed thank you.

**Reply by bitoandy** [reply]
@MuhammadMahmoudH , can you also drop a note on support@bito.ai with the screenshots and any details to see the issue you are facing?
Also provide the details of PHPStorm version and Bito IDE version you are using on the support email.

**Reply by MuhammadMahmoudH** [reply]
Ok thanks

**Comment by denisbetsi** [comment]
Few issues after using this
1) I can't stop the process of Indexing once it starts
2) No way to exclude folders from being indexed, so now it's stuck trying to process virt folder or about 16.6K files, it's been going for about 14 hours  and it's at 54% only while my CPU on the laptop is burning hot
3) See point #1, no way to stop the process, so I have no idea how to decouple this process

**Reply by denisbetsi** [reply]
Never mind, found a way to stop it, had to hit Delete button, would be nice to be able to Pause it.

**Reply by bitoandy** [reply]
Hi @denisbetsi ,

We are working on adding ignore functionality similar to git ignore in the next update for indexing.

**Reply by denisbetsi** [reply]
May I suggest to also default to .gitignore if your own ignore isn't present, this way it will not require double ignore files.

**Reply by bitoandy** [reply]
Yes, gitignore will be used to ignore files and if you want to ignore files beyond that without modifying gitignore only then you will need to use Bito’s ignore file

**Comment by zvalcav** [comment]
Some debug output would be nice. One of my repositories' indexing freezes at exactly same amount of indexed files and I'm unable to get past it. Problem is I don't know which file is causing it so I can't .bitoignore it. In VSCode it could be placed for example here:

![image](https://github.com/gitbito/bitoai/assets/8685504/9e493920-cbe5-45fe-ace1-59036c511c66)

Since v1.1.7 version of Bito I'm suspecting it from being heavy on CPU. But I can't prove it yet. Just a suspicion ATM.

**Comment by hkkzzxz** [comment]
I am unable to upload my local code and cannot find the button
![image](https://github.com/gitbito/bitoai/assets/80557578/9cb41abf-f6bd-4af2-92b7-85208c4aa2f2)

**Reply by lcuevastodoit** [reply]
Hi, when you are premium, you can see a warning if you want to enable "AI understand your code", when you say "yes", in that moment begins the indexation of the current app. I did notice that you should be need a .bitoignore file in the app`s root folder before enable the indexing process, so it skip unnecessary folders like bin, src, node_modules, logs, etc. Because if the total size of the app files (after .bitoignore skipping)  is greater than 150MB approx, you will receive a warning that indexing is not supported. In that case, just add the .bitoignore, delete the "repo icon" on "Manage indexed repos" section, and after that click again in "Start indexing"

This is an example of .bitoignore content

```text
tests/
app/assets/images/
public/
.git/
data/
pruebas/
**/target/
**/target-eclipse/
**/web-app/images/
**/lib/
```

**Reply by hkkzzxz** [reply]
Yes, I confirm that I have opened a membership and it has not yet expired, but Bito has not upgraded my file loading
![image](https://github.com/gitbito/bitoai/assets/80557578/995b2de6-8cfd-424b-8d78-6ae7716c6dae)

**Reply by bitoandy** [reply]
Hi @hkkzzxz ,

Regret the delay in responding.
Can you do the following:
1. Close and exit the IDE.
2. In the user home directory there is a .bito folder, within that index is maintained in folder "localcodesearch".
3. Delete the "localcodesearch" folder.
4. Restart the IDE and try indexing again/ go to manage repos tab, you should be able to see the folders and manage indexing.
